documentation/architecture/decisions/0003-crate-feature-flags.md

```diff
@@ -104,3 +147,5 @@ The rules of use are as follows:
 2. otherwise, use `ockam_core::compat::<mod>` equivalents where available. (e.g. `std::sync::Arc -> ockam_core::compat::sync::Arc`)
 3. if you need to add new items to compat, follow the originating namespace. (e.g. `compat::vec::Vec` and not `compat::Vec`)
 4. if none of the above apply use the conditional compilation attributes as documented above
+
+In the future, we may move `ockam_core::compat` into a dedicated crate in order to reduce feature fragility, however prior to general availability, this is probably fine).
```

Comment:
Thinking about this more, it wouldn't really solve the problem I'm concerned with. Essentially, having pre-1.0 public dependencies prevents us from becoming 1.0 in a sense, since we have a different stability promise/commitment than our public dependencies. (see https://rust-lang.github.io/api-guidelines/necessities.html#c-stable for some rationalization here)